activesupport/lib/active_support/test_case.rb

```diff
@@ -91,6 +92,10 @@ def parallelize(workers: :number_of_processors, with: :processes)
         Minitest.parallel_executor = executor
 
         parallelize_me!
+
+        # ActiveSupport.test_order is shadowing Minitest::Test.test_order
+        # It requires to be updated by hand.
```

Comment:
I don't quite understand the why behind this comment. Can you elaborate?

Reply:
Somehow the reader `test_order` defined [here](https://github.com/rails/rails/blob/1baa0684a06199203f14ac696c2d3ef9042853f2/activesupport/lib/active_support/test_case.rb#L41-L43) is not overridden after calling `parallelize_me!` which extends from [this](https://github.com/seattlerb/minitest/blob/3c6576a51f4e266996e3459d7a0dd054eb4c87f7/lib/minitest/parallel.rb#L59-L67) module. I can't determine why. 

The reader uses `ActiveSupport.test_order`, which never gets updated to use new value `:parallel`. I think it's important for consistency that the reader reflects the correct test order. I'm also relying on this in my tests.

Reply:
@eileencodes

Following up on this, I found that in some cases where parallelization was enabled `test_order` was returning `:parallel`, whereas in others it returned `:random`.

I discovered that it was because the method `ActiveRecord::TestCase.test_order` overrides the one from `Minitest`, and in some cases with inheritance (using a base class that calls `parallelize`), it ends up having more prevalence over the `Minitest` one (which just returns `:parallel`).

It's not a big deal as the library handles both cases `:parallel` and `:random` equally (examples are sorted randomly), but consistency would be great. I included a short comment explaining why the manual update is required, I can move to the commit message if that's better.

In case you're curious, applying following diff in master and running a sample test file

```diff
diff --git a/activesupport/test/abstract_unit.rb b/activesupport/test/abstract_unit.rb
index c349d16688..f3d69de3bf 100644
--- a/activesupport/test/abstract_unit.rb
+++ b/activesupport/test/abstract_unit.rb
@@ -34,6 +34,8 @@ class ActiveSupport::TestCase
     parallelize(with: :threads)
   end
 
+  p singleton_class.ancestors # 1
+
   include ActiveSupport::Testing::MethodCallAssertions
 
   private
diff --git a/activesupport/test/core_ext/duration_test.rb b/activesupport/test/core_ext/duration_test.rb
index cb92781680..d357d461d1 100644
--- a/activesupport/test/core_ext/duration_test.rb
+++ b/activesupport/test/core_ext/duration_test.rb
@@ -10,6 +10,8 @@
 class DurationTest < ActiveSupport::TestCase
   include TimeZoneTestHelpers
 
+  p singleton_class.ancestors # 2
+
   def test_is_a
     d = 1.day
     assert d.is_a?(ActiveSupport::Duration)
```


```
activesupport $ bin/test test/core_ext/duration_test.rb
```

Results in the following:

Ancestors chain in 1:

```
[#<Class:ActiveSupport::TestCase>, Minitest::Parallel::Test::ClassMethods, ... shortened for readability]
```

Ancestors chain in 2:
```
[#<Class:DurationTest>, #<Class:ActiveSupport::TestCase>, Minitest::Parallel::Test::ClassMethods, ... shortened for readability]
```

When a test class inherits from a base class that enabled parallelization, the `test_order` method that it uses will be the one from `ActiveSupport::TestCase` because it's closer in the ancestors chain.